homeassistant/components/escea/climate.py

```diff
@@ -0,0 +1,218 @@
+"""Support for the Escea Fireplace."""
+from __future__ import annotations
+
+from collections.abc import Coroutine
+import logging
+from typing import Any
+
+from pescea import Controller
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_LOW,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, PRECISION_WHOLE, TEMP_CELSIUS
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    DATA_DISCOVERY_SERVICE,
+    DISPATCH_CONTROLLER_DISCONNECTED,
+    DISPATCH_CONTROLLER_DISCOVERED,
+    DISPATCH_CONTROLLER_RECONNECTED,
+    DISPATCH_CONTROLLER_UPDATE,
+    ESCEA,
+    ESCEA_MANUFACTURER,
+    ICON,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+_ESCEA_FAN_TO_HA = {
+    Controller.Fan.FLAME_EFFECT: FAN_LOW,
+    Controller.Fan.FAN_BOOST: FAN_HIGH,
+    Controller.Fan.AUTO: FAN_AUTO,
+}
+_HA_FAN_TO_ESCEA = {v: k for k, v in _ESCEA_FAN_TO_HA.items()}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, config: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Initialize an Escea Controller."""
+    disco = hass.data[DATA_DISCOVERY_SERVICE]
+
+    @callback
+    def init_controller(ctrl: Controller) -> None:
+        """Register the controller device."""
+
+        _LOGGER.debug("Controller UID=%s discovered", ctrl.device_uid)
+
+        device = ControllerDevice(ctrl)
+        async_add_entities([device])
+
+    # create any components not yet created
+    for controller in disco.pi_disco.controllers.values():
+        init_controller(controller)
+
+    # connect to register any further components
+    config.async_on_unload(
+        async_dispatcher_connect(hass, DISPATCH_CONTROLLER_DISCOVERED, init_controller)
+    )
+
+
+class ControllerDevice(ClimateEntity):
+    """Representation of Escea Controller."""
+
+    _attr_fan_modes = list(_HA_FAN_TO_ESCEA)
+    _attr_hvac_modes = [HVACMode.HEAT, HVACMode.OFF]
+    _attr_icon = ICON
+    _attr_precision = PRECISION_WHOLE
+    _attr_should_poll = False
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.FAN_MODE
+    )
+    _attr_target_temperature_step = PRECISION_WHOLE
+    _attr_temperature_unit = TEMP_CELSIUS
+
+    def __init__(self, controller: Controller) -> None:
+        """Initialise ControllerDevice."""
+        self._controller = controller
+
+        self._attr_min_temp = controller.min_temp
+        self._attr_max_temp = controller.max_temp
+
+        self._attr_unique_id = controller.device_uid
+        self._attr_name = f"Escea Fireplace {self._attr_unique_id}"
```

Comment:
Please see our new entity naming guidelines and set `_attr_has_entity_name` to `True`:
https://developers.home-assistant.io/docs/core/entity#entity-naming

Reply:
Thanks, ok, I think I get how that works now. I've set the _attr_has_entity_name, then I haven't assigned the _attr_name, as my understanding is the friendly name is generated from the device name in that instance. Hopefully that's right... but will test it locally to check my understanding.